ROADMAP.md

```diff
@@ -0,0 +1,154 @@
+I imagine some kind of OpenAL like API, supporting spatial sound source objects, audio streaming, attenuation, etc.
+Provide low-level access?
+
+** Next Steps**
+
+### Serialization/Reflection
+Type reflection offers additional information about the type, which is available at compile- or run-time. In comparison to C++, to can be achieved easier with Rusts macro system.
+In our engine we want to couple the ECS with serialization/reflection to easily load/save game scenes from various formats. Additionally, it helps us in to build safe network protocols, writing script and GUI bindings.
+
+** Next Steps**
+ * Integrate serde based reflection
+ * Serialize/Deserialize ecs worlds
+
+### Streaming/Asset Handling
+TODO
+
+**Parallelism** TODO: Load assets in parallel
+**Data-Driven** TODO: low loading overhead
+
+** Next Steps**
+ * Improve current approach regarding robustness (error handling) and flexibility with scene formats
+ * Implement loaders for common file formats and archives
+
+### Documentation/Tutorial
+TODO
+
+** Next Steps**
+
+### Physics
+TODO: I would consider physics pretty difficult part to do *right*, therefore not part of the short term goals.
+Maybe rely on external libraries first.
+Should still provide ways for collision detection, allowing to do collision checks, raycasts, etc. (-> ncollide?).
+
+**Parallelism** TODO: internal physic parallelism, possible GPU implementation (hard), difficult with external libraries
+**Data-Driven** TODO: integration issue: physics world <-> ECS world, having 2 different representations? if so how to keep them in sync and integrate with other conflicting processors (e.g spatial hierarchy)
+
+** Next Steps**
+ * Integrate an existing library or write from scratch (requires further planning then)
+ * Provide collision testing API (triggers, raycasts, etc.)
+
+### Tooling
+We want to focus on developing a game engine, even though a rich toolset is required to simplify different task for users. If possible, we can integrate existing tools like profilers and 3D editors.
+'In-house' tools would be implemented on demand, e.g. engine specific data formats, own editor (Qt based?).
+
+** Next Steps**
+ * Integrate better logging (high priority -> short-term)
+
+### Scripting
+Possible scripting languages: mruby, lua, moonscript, dyon, ..
+
+Supporting a scripting language would support rapid-prototyping and faster iteration cycles. Also would be more attractive for developers who are not so much familiar with Rust.
+Scripting could be use for several subsystems like animation, particle system, AI, etc.
+
+** Next Steps**
+ * Evaluate the different scripting languages with respect to interoperability, performance, usability, ..
+ 
+### Graphical UI
+** Next Steps**
+ * Compatibility of GUIs with entity component systems (UI elements as entities <-> only root UI elements)
+ * Integrate an existing UI library (e.g conrod) or develop a new library from scratch (requires further planning then)
```

Comment:
the op commented in order levereage web tecnology for gui it would be better to use servo , as it is more ligthweight than chormiun , however the projects that would have allowed to utilize it as a framework or library , instead of an embedded browser havent materialized  , as i understand it , the  way to interact with it 3for the near future,   will be by using the cef  api ; finally , it is now expected to run in multiple processes ( again , just as chorme does ) , which could hurt performance 

All